### PR TITLE
INSTALL-1063: Stabilize 6.1.2 branch

### PR DESCRIPTION
**Pull request by lulivi**
### Proposed changes

Create a warning quality gate.

### CODE REVIEW

File: resources/ci/buildConfiguration.groovy [hunk 1]
```diff
@@ -139,12 +143,12 @@ pipeline {
             matrix {
                 axes {
                     axis {
-                        name 'buildMode'
-                        values 'release', 'debug'
+                        name 'buildType'
+                        values 'release'  // TODO: Recover debug build type
```
Comment (lulivi) [inline resources/ci/buildConfiguration.groovy:147]: Bring back once the tests are finished

File: resources/ci/buildConfiguration.groovy [hunk 2]
```diff
@@ -139,12 +143,12 @@ pipeline {
...
                     }
                     axis {
                         name 'linkMode'
-                        values 'static', 'dynamic'
+                        values 'dynamic'  // TODO: Recover static link mode
```
Comment (lulivi) [inline resources/ci/buildConfiguration.groovy:151]: Bring back once the tests are finished

File: Jenkinsfile [hunk 3]
```diff
@@ -58,17 +62,25 @@ Map branchJobs(String[] branches) {
...
             steps {
+                checkoutCommunityRepoBranch(
+                    'rticonnextdds-cmake-utils', env.BRANCH_NAME
+                )
                 script {
-                    String[] examplesBranches = readYaml(
-                        file: "${env.WORKSPACE}/resources/ci/config.yaml"
-                    ).versions.keySet()
-                    parallel branchJobs(examplesBranches)
+                    CONNEXT_VERSIONS.addAll(
```
Comment (ManuelJNunez) [inline Jenkinsfile:79]: CONNEXT_VERSIONS contains the branches to build, right?

File: resources/ci/buildArchitectureConfigurations.groovy [hunk 4]
```diff
@@ -49,15 +55,24 @@ void runBuildConfigurationJob(
...
 }
 
+/**
+ * Remove the branch prefix from the Connext version.
+ *
+ * @param repositoryBranch The rticonnextdds-examples branch to build.
+ * @returns The Connext version to build.
+ */
+String getVersionFromBranch(String repositoryBranch) {
+    return repositoryBranch - 'release/'
```
Comment (ManuelJNunez) [inline resources/ci/buildArchitectureConfigurations.groovy:65]: I think that you can store just the version in the YAML file.

Comment (ManuelJNunez) [reply resources/ci/buildArchitectureConfigurations.groovy:65]: and then you add the release/ instead of removing it.

File: Jenkinsfile (hunk 3 above)
Comment (lulivi) [reply Jenkinsfile:79]: Yes!

File: resources/ci/config.yaml [hunk 5]
```diff
@@ -1,13 +1,13 @@
 versions:
   master:
-  #   linux: x64Linux3gcc4.8.2
+    # linux: x64Linux3gcc4.8.2
```
[the remark this answers is not in the document]
Comment (lulivi) [reply resources/ci/config.yaml:3]: Yes! Currenlty we use the x64Linux4gcc7.3.0 for all the versions

File: resources/ci/buildArchitectureConfigurations.groovy (hunk 4 above)
Comment (lulivi) [reply resources/ci/buildArchitectureConfigurations.groovy:65]: This is needed in order to know from the yaml, which of the branch names are release to pass it to the job.

**Review by ManuelJNunez — APPROVED** [review]

**Pull request merged**